### PR TITLE
Update Jenkins CI

### PR DESCRIPTION
**Pull request by dalg24**
Cleaning up the Dockerfile and changing the way CI is setup to use a Jenkinsfile.

### CODE REVIEW

**Comment by dalg24** [comment]
Ready for review

**Comment by dalg24** [comment]
I disabled the old CI build so we need to drop the requirement on `continuous-integration/jenkin-ci/pr` check passing.

File: scripts/jenkins/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
-FROM nvidia/cuda:10.1-devel-ubuntu18.04
+ARG BASE=nvidia/cuda:10.2-devel
```
Comment (sslattery) [inline scripts/jenkins/Dockerfile]: Above says `stage('CUDA-10.1-NVCC-DEBUG')` but this is 10.2?

Comment (dalg24) [reply scripts/jenkins/Dockerfile]: Typo

**Pull request merged**